thavalon-webapp/src/components/gameComponents/playerBoard.tsx

```diff
@@ -104,7 +104,14 @@ export function PlayerBoard(): JSX.Element {
      * @param message The GameMessage from the server
      */
     function handleGameMessage(message: GameMessage): void {
-        if (message.messageType !== GameMessageType.Error && message.messageType !== GameMessageType.Toast) {
+        // TODO: This has become a steaming pile of spaghetti and a large source of bugs.
```

Comment:
Would a quick fix for now be to let `mapMessageToGamePhase` return `undefined` if the message doesn't correspond to a phase, and then skip the `setGamePhase` call in that case?

Reply:
Done